awx/ui/client/features/jobs/jobsList.controller.js

```diff
@@ -40,6 +41,63 @@ function ListJobsController (
     vm.job_dataset = Dataset.data;
     vm.jobs = Dataset.data.results;
 
+    $scope.$watch('$state.params', () => {
+        setToolbarSort();
+    }, true);
+
+    function setToolbarSort () {
+        const orderByValue = _.get($state.params, 'job_search.order_by');
+        const sortValue = _.find(vm.toolbarSortOptions, (option) => option.value === orderByValue);
+        if (sortValue) {
+            vm.toolbarSortValue = sortValue;
+        } else {
+            vm.toolbarSortValue = toolbarSortDefault;
```

Comment:
you use this after you set it, I think you need to switch this and the const declaration